src/status_line_manager.ts

```diff
@@ -33,28 +31,38 @@ export class StatusLineManager implements Disposable {
         this.statusBar = window.createStatusBarItem(StatusBarAlignment.Left, -10);
         this.statusBar.show();
         this.disposables.push(this.statusBar, eventBus.on("redraw", this.handleRedraw, this));
-        actions.add("refresh-status-items", (items: string[]) => this.setStatus(items, StatusType.Custom));
-    }
 
-    private setStatus(status: string[], type: StatusType.Custom): void;
-    private setStatus(status: string, type: Omit<StatusType, "Custom">): void;
-    private setStatus(status: string | string[], type: StatusType): void {
-        if (Array.isArray(status)) {
-            this._customStatus = status;
-        } else {
-            switch (type) {
-                case StatusType.Mode:
-                    this._modeText = status;
-                    break;
-                case StatusType.Cmd:
-                    this._cmdText = status;
-                    break;
-                case StatusType.Msg:
-                    this._msgText = status;
-                    break;
+        const refreshNvimStatusLineTimer = setInterval(async () => {
+            let sl = (await this.client.lua(`
+            if vim.o.laststatus == 0 then return "" end
+            return vim.api.nvim_eval_statusline(vim.o.statusline, {}).str
```

Comment:
'statusline' is blank by default, which then uses Nvim's internal statusline logic. That's a historical quirk of Vim, not really up to vscode-neovim to work around it. Tracking issue: https://github.com/neovim/neovim/issues/28444

Reply:
> When 'statusline' option is blank, Nvim still shows a statusline. In that case, the vscode statusline will be blank even though Nvim's _rendered_ statusline is not blank.

I thought about this, I don't think anyone needs this default statusline

> OTOH, it's probably good that users can disable this vscode-neovim behavior by setting 'statusline' to empty

Yes. So I removed the settings for the laststatus and statusline options instead of setting a default value :)